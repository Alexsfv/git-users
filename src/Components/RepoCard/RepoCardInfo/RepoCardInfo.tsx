import { observer } from 'mobx-react'
import React from 'react'

type RepoCardInfoProps = {
    title: string
    data: string | null | number
}
const RepoCardInfo: React.FC<RepoCardInfoProps> = observer(({title, data}) => {
    const stringData = data + ''
    const isHidden = stringData === 'false' || stringData === 'null' || stringData === 'undefined' || stringData === ''

    if (!isHidden) {
        return (
            <div className="w-100 mb-2 bg-light shadow-sm rounded">
                <p className="mt-0 mb-1 font-weight-bold text-center">{title}</p>
                <p className="ml-1 card-text text-center">{data}</p>
            </div>
        )
    }

    return null
})

export default RepoCardInfo