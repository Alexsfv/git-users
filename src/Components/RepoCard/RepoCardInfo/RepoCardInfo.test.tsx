import { shallow } from 'enzyme'
import React from 'react'
import RepoCardInfo from './RepoCardInfo'

describe("RepoCardInfo component", () => {
    it("should render RepoCardInfo", () => {
        const component = shallow(<RepoCardInfo title="some title" data="11-22-33"/>)
        expect(component).toMatchSnapshot()
    })

    it("should empty render RepoCardInfo", () => {
        const component = shallow(<RepoCardInfo title="some title" data=""/>)
        expect(component).toMatchSnapshot()
    })
})