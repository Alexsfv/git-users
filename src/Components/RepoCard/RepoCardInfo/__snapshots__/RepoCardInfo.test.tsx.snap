// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoCardInfo component should empty render RepoCardInfo 1`] = `""`;

exports[`RepoCardInfo component should render RepoCardInfo 1`] = `
<div
  className="w-100 mb-2 bg-light shadow-sm rounded"
>
  <p
    className="mt-0 mb-1 font-weight-bold text-center"
  >
    some title
  </p>
  <p
    className="ml-1 card-text text-center"
  >
    11-22-33
  </p>
</div>
`;
